include/gsl/span

```diff
@@ -121,139 +122,165 @@ namespace details
     {
     };
 
-    template <class Span, bool IsConst>
+    template <class type>
     class span_iterator
     {
-        using element_type_ = typename Span::element_type;
-
     public:
-#ifdef _MSC_VER
-        // Tell Microsoft standard library that span_iterators are checked.
-        using _Unchecked_type = typename Span::pointer;
-#endif
-
         using iterator_category = std::random_access_iterator_tag;
-        using value_type = std::remove_cv_t<element_type_>;
-        using difference_type = typename Span::index_type;
-
-        using reference = std::conditional_t<IsConst, const element_type_, element_type_>&;
-        using pointer = std::add_pointer_t<reference>;
+        using value_type = std::remove_cv_t<type>;
+        using difference_type = ptrdiff_t;
+        using pointer = std::add_pointer_t<type>;
+        using reference = std::add_lvalue_reference_t<type>;
 
-        span_iterator() = default;
-
-        constexpr span_iterator(const Span* span, difference_type idx) noexcept
-            : span_(span), index_(idx)
-        {}
-
-        friend span_iterator<Span, true>;
-        template <bool B, std::enable_if_t<!B && IsConst>* = nullptr>
-        constexpr span_iterator(const span_iterator<Span, B>& other) noexcept
-            : span_iterator(other.span_, other.index_)
-        {}
-
-        GSL_SUPPRESS(bounds.1) // NO-FORMAT: attribute
-        constexpr reference operator*() const
+#ifdef _MSC_VER
+        using _Unchecked_type = typename pointer;
```

Comment:
Belated response to your email question: the presence of `_Unchecked_type` tells an older STL (IIRC VS2017-era) that this iterator type participates in our checking and unwrapping machinery. Current STLs don't use it anymore, but STLs that you support do.

Reply:
I will leave it then, thanks!